crates/uv/src/commands/pip_compile.rs

```diff
@@ -347,18 +348,21 @@ pub(crate) async fn pip_compile(
             "{}",
             "# This file was autogenerated by uv via the following command:".green()
         )?;
-        writeln!(
-            writer,
-            "{}",
-            format!(
-                "#    uv {}",
-                env::args_os()
-                    .skip(1)
-                    .map(|arg| arg.normalized_display().to_string())
-                    .join(" ")
+        let mut arguments = env::args_os()
+            .skip(1)
+            .map(|arg| arg.normalized_display().to_string())
+            .join(" ");
+        if !include_index_url {
+            // Matches and captures --extra-index-url or --index-url along with their values,
+            // accounting for both quoted and unquoted values, and removes any trailing whitespace.
+            // SAFETY: The regex pattern is a constant string and is valid.
+            let pattern = Regex::new(
+                r#"(--extra-index-url|--index-url)[\s=]+("[^"\s]+"|'[^'\s]+'|[^\s]+)\s*"#,
             )
```

Comment:
Does it make sense to use a regex here? Shouldn't we just do this filtering above when we construct the arguments to display?

Reply:
I started with this approach, but it looks hard for me to work with `env::args_os()` to do filtering - arguments could come both together (`--index-url=url`) and separately (`--index-url url`). But if you know the way to do this without regex here - I'll be more than happy to change my approach! I hate regexps 😅

Reply:
I guess when we see `--index-url` we'd omit it and any subsequent arguments until we see another that starts with `-`, I think that should be safe? I admit I'm not entirely certain which approach is better.

Reply:
Unfortunately, we can't omit any subsequent arguments till "-" because it could be situation with `uv pip compile --index-url https://pypi.org/simple requirements.in --some-other-arg`, but maybe it's safe to exclude the next one after index command itself...

Reply:
Unfortunately, we can't omit any subsequent arguments till "-" because it could be situation with `uv pip compile --index-url https://pypi.org/simple requirements.in --some-other-arg`, but maybe it's safe to exclude the next one after index command itself...

Reply:
So, I came up with this solution if we want to exclude regular expression. Me personally don't like it, but it's working. In the future, if we also want to exclude find link through other condition, that could also be done this way

```rust
if include_header {
    writeln!(
        writer,
        "{}",
        "# This file was autogenerated by uv via the following command:".green()
    )?;
    let mut arguments = String::new();
    let mut skip_next = false;
    for arg in env::args_os().skip(1) {
        let arg = arg.normalized_display().to_string();
        if skip_next {
            skip_next = false;
            continue;
        }

        if !include_index_url {
            if arg == "--extra-index-url" || arg == "--index-url" {
                skip_next = true;
                continue;
            } else if arg.starts_with("--extra-index-url=") || arg.starts_with("--index-url=") {
                continue;
            }
        }
        arguments.push_str(&arg.normalized_display().to_string());
        arguments.push(' ');
    }
    writeln!(
        writer,
        "{}",
        format!("#    uv {}", arguments.trim()).green()
    )?;
}
```

Reply:
Actually, during the discussion with my friend, we come up with some rustic way of filtering arguments. Regex removed and replaced with conditional filtering of arguments 👍